ui/app/helpers/constants/transactions.js

```diff
@@ -7,14 +7,15 @@ export const CONFIRMED_STATUS = 'confirmed'
 export const FAILED_STATUS = 'failed'
 export const DROPPED_STATUS = 'dropped'
 export const CANCELLED_STATUS = 'cancelled'
+export const PENDING_STATUS = 'pending'
 
 export const TOKEN_METHOD_TRANSFER = 'transfer'
 export const TOKEN_METHOD_APPROVE = 'approve'
 export const TOKEN_METHOD_TRANSFER_FROM = 'transferfrom'
 
 export const SEND_ETHER_ACTION_KEY = 'sentEther'
 export const DEPLOY_CONTRACT_ACTION_KEY = 'contractDeployment'
-export const APPROVE_ACTION_KEY = 'approve'
```

Comment:
@Gudahtt did this before I got more comfortable with the internationalization system -- though I'm still torn here. The text for the approve message is changing from 'Approve' to 'Approve Spending Limit' -- it seems like changing the text would be quite the break from the current translations but it would still capture the essence of the message ... should I just rephrase the en/messages.json "message" for the 'approve' key instead of adding a new key?

Reply:
Hmm. Yeah that seems fine to me 🤔 

I wish we had a better way of handling changes to these messages. Leaving the "Approve" transactions does seem better than erasing them though.

Reply:
approve action key restored to 'approve'
https://github.com/MetaMask/metamask-extension/blob/b695366e0c978cd969e86ba38eb8a7396b2c8daf/ui/app/helpers/constants/transactions.js#L36